middleware/signmigration.py

```diff
@@ -0,0 +1,246 @@
+    sk = ecdsa.SigningKey.from_string(
+        bytes.fromhex(options.key),
+        curve=ecdsa.SECP256k1
+    )
+    signature = sk.sign_digest(
+        migration_spec.get_authorization_digest(),
+        sigencode=ecdsa.util.sigencode_der
+    )
+    # Add the signature to the authorization and save it to disk
+    sgx_authorization.add_signature(signature.hex())
+    sgx_authorization.save_to_jsonfile(options.output_path)
+    info(f"SGX migration authorization saved to {options.output_path}")
+
+
+def do_eth(options):
+    _require_output_path(options)
+    if options.path is None:
+        options.path = DEFAULT_ETH_PATH
+    # Parse path
+    path = BIP32Path(options.path)
+    eth = None
+    try:
+        # Get dongle access (must have ethereum app open)
+        eth = get_eth_dongle(options.verbose)
+        # Retrieve public key
+        info(f"Retrieving public key for path '{str(path)}'...")
+        pubkey = eth.get_pubkey(path)
+        info(f"Public key: {pubkey.hex()}")
+
+        # If options.pubkey is True, we just want to retrieve the public key
+        if options.pubkey:
+            info(f"Opening public key file {options.output_path}...")
+            info("Adding public key...")
+            with open(options.output_path, "w") as file:
+                file.write("%s\n" % pubkey.hex())
+            info(f"Public key saved to {options.output_path}")
+            return
+
+        # Is there an existing migration authorization? Read it
+        sgx_authorization = None
+        if not isfile(options.output_path):
+            raise AdminError("Invalid SGX migration authorization file: "
+                             f"{options.output_path}")
+
+        info(f"Opening SGX migration authorization file {options.output_path}...")
+        sgx_authorization = SGXMigrationAuthorization.from_jsonfile(options.output_path)
+        migration_spec = sgx_authorization.migration_spec
+        info("Signing with dongle...")
+        try:
+            signature = eth.sign(path, migration_spec.msg.encode('ascii'))
+            vkey = ecdsa.VerifyingKey.from_string(pubkey, curve=ecdsa.SECP256k1)
+
+            if not vkey.verify_digest(
+                signature, migration_spec.get_authorization_digest(),
+                sigdecode=ecdsa.util.sigdecode_der
+            ):
+                raise Exception()
+        except Exception:
+            raise AdminError(f"Bad signature from dongle! (got '{signature.hex()}')")
+        # Add the signature to the authorization and save it to disk
+        sgx_authorization.add_signature(signature.hex())
+        sgx_authorization.save_to_jsonfile(options.output_path)
+        info(f"SGX migration authorization saved to {options.output_path}")
+    except AdminError:
+        raise
+    except Exception as e:
+        raise AdminError(f"Error signing with dongle: {e}")
+    finally:
+        dispose_eth_dongle(eth)
+
+
+def main():
+    logging.disable(logging.CRITICAL)
```

Comment:
Why are we doing this?

Reply:
TBH, I just copied this from [signapp.py](https://github.com/rsksmart/rsk-powhsm/blob/master/middleware/signapp.py#L50).

Now that you mentioned, this apparently has no effect, since we're logging using `misc.info`. Should this be removed (probably from signapp.py as well)?

Reply:
sounds good. could you please make sure the output and behaviour remain unaltered after the changes? thanks!